app/Listeners/MetricsListener.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * SPDX-License-Identifier: MIT
+ * Copyright (c) 2017-2018 Tobias Reich
+ * Copyright (c) 2018-2025 LycheeOrg.
+ */
+
+namespace App\Listeners;
+
+use App\Events\Metrics\AlbumDownload;
+use App\Events\Metrics\AlbumShared;
+use App\Events\Metrics\AlbumVisit;
+use App\Events\Metrics\PhotoDownload;
+use App\Events\Metrics\PhotoFavourite;
+use App\Events\Metrics\PhotoShared;
+use App\Events\Metrics\PhotoVisit;
+use App\Models\Configs;
+use Illuminate\Support\Facades\DB;
+
+/**
+ * Just logging of Cache events.
+ */
+class MetricsListener
+{
+	/**
+	 * Handle the event.
+	 */
+	public function handle(AlbumDownload|AlbumShared|AlbumVisit|PhotoDownload|PhotoFavourite|PhotoShared|PhotoVisit $event): void
```

Comment:
my above comment will save modifying this parameter type every time we add a new event ;)

Reply:
WTRC: [a4a4e18](https://github.com/LycheeOrg/Lychee/pull/3228/commits/a4a4e18a59583d98569932f5bb11a7eea3137965)